compiler/hash-typecheck/src/traverse/visitor.rs

```diff
@@ -1640,80 +1654,65 @@ impl<'tc> visitor::AstVisitor for TcVisitor<'tc> {
 
     fn visit_assign_op_expr(
         &mut self,
-        _ctx: &Self::Ctx,
-        _node: hash_ast::ast::AstNodeRef<hash_ast::ast::AssignOpExpr>,
+        ctx: &Self::Ctx,
+        node: hash_ast::ast::AstNodeRef<hash_ast::ast::AssignOpExpr>,
     ) -> Result<Self::AssignOpExprRet, Self::Error> {
-        todo!()
-    }
+        debug_assert!(node.operator.is_re_assignable());
 
-    type BinaryExprRet = TermId;
+        let rhs = self.visit_expr(ctx, node.rhs.ast_ref())?;
 
-    fn visit_binary_expr(
-        &mut self,
-        ctx: &Self::Ctx,
-        node: hash_ast::ast::AstNodeRef<hash_ast::ast::BinaryExpr>,
-    ) -> Result<Self::BinaryExprRet, Self::Error> {
-        let walk::BinaryExpr { lhs, rhs, .. } = walk::walk_binary_expr(self, ctx, node)?;
+        let name = match node.lhs.kind() {
+            ast::ExprKind::Variable(name) => name.name.ident,
+            // @@Incomplete: what about tuples, or array indices?
+            _ => {
+                return Err(TcError::InvalidAssignSubject {
+                    location: self.source_location_at_node(node).into(),
+                });
+            }
+        };
 
-        let operator_fn = |trait_fn_name: &str| {
-            let prop_access = self.builder().create_prop_access(lhs, trait_fn_name);
-            self.copy_location_from_node_to_target(node.operator.ast_ref(), prop_access);
+        let var_term = self.builder().create_var_term(name);
+        self.copy_location_from_node_to_target(node, var_term);
 
-            let builder = self.builder();
-            builder.create_fn_call_term(
-                prop_access,
-                builder.create_args([builder.create_nameless_arg(rhs)], ParamOrigin::Fn),
-            )
-        };
+        let ScopeMember { member, scope_id, index } =
+            self.scope_manager().resolve_name_in_scopes(name, var_term)?;
 
-        let lazy_operator_fn = |visitor: &mut Self, trait_name: &str| -> TcResult<TermId> {
-            let lhs_ty = visitor.typer().infer_ty_of_term(lhs)?;
-            let rhs_ty = visitor.typer().infer_ty_of_term(rhs)?;
+        // We want to create the type that represents this operation, and then
+        // ensure that it typechecks...
+        let ty = self.create_operator_fn(member.ty(), rhs, node.operator.ast_ref(), true);
```

Comment:
Why are you getting its type? Aren't you meant to get the member's value?

Reply:
yes this is wrong, can we assume that `.value()` will always yield a value?

Reply:
No, but you can just pass in the member using `Var` term.